PWGHF/Tasks/taskHFCorrelations.cxx

```diff
@@ -0,0 +1,505 @@
+  Configurable<bool> processHFHadrons{"processHFHadrons", "false", "Flag to process HF-h correlations"};
+
+  Configurable<int> cfgNoMixedEvents{"cfgNoMixedEvents", 5, "Number of mixed events per event"};
+
+  //  configurables for collisions
+  Configurable<float> cfgCutVertex{"cfgCutVertex", 7.0f, "Accepted z-vertex range"};
+
+  //  configurables for associated particles
+  Configurable<float> cfgCutEta{"cfgCutEta", 0.8f, "Eta range for tracks"};
+  Configurable<float> cfgCutPt{"cfgCutPt", 0.5f, "Minimum pT for tracks"};
+
+  //  configurables for HF candidates
+  Configurable<int> d_selectionFlagD0{"d_selectionFlagD0", 1, "Selection Flag for D0"};
+  Configurable<int> d_selectionFlagD0bar{"d_selectionFlagD0bar", 1, "Selection Flag for D0bar"};
+  Configurable<double> cutYCandMax{"cutYCandMax", -1., "max. cand. rapidity"};
+  Configurable<std::vector<double>> bins{"pTBins", std::vector<double>{hf_cuts_d0_topik::pTBins_v}, "pT bin limits"};
+
+  //  configurables for containers
+  ConfigurableAxis axisVertex{"axisVertex", {14, -7, 7}, "vertex axis for histograms"};
+  ConfigurableAxis axisDeltaPhi{"axisDeltaPhi", {72, -PIHalf, PIHalf * 3}, "delta phi axis for histograms"};
+  ConfigurableAxis axisDeltaEta{"axisDeltaEta", {48, -2.4, 2.4}, "delta eta axis for histograms"};
+  ConfigurableAxis axisPtTrigger{"axisPtTrigger", {VARIABLE_WIDTH, 0.5, 1.0, 1.5, 2.0, 3.0, 4.0, 6.0, 10.0}, "pt trigger axis for histograms"};
+  ConfigurableAxis axisPtAssoc{"axisPtAssoc", {VARIABLE_WIDTH, 0.5, 1.0, 1.5, 2.0, 3.0, 4.0, 6.0}, "pt associated axis for histograms"};
+  ConfigurableAxis axisMultiplicity{"axisMultiplicity", {VARIABLE_WIDTH, 0, 5, 10, 20, 30, 40, 50, 100.1}, "multiplicity axis for histograms"};
+  ConfigurableAxis axisVertexEfficiency{"axisVertexEfficiency", {10, -10, 10}, "vertex axis for efficiency histograms"};
+  ConfigurableAxis axisEtaEfficiency{"axisEtaEfficiency", {20, -1.0, 1.0}, "eta axis for efficiency histograms"};
+  ConfigurableAxis axisPtEfficiency{"axisPtEfficiency", {VARIABLE_WIDTH, 0.5, 0.6, 0.7, 0.8, 0.9, 1.0, 1.25, 1.5, 1.75, 2.0, 2.25, 2.5, 2.75, 3.0, 3.25, 3.5, 3.75, 4.0, 4.5, 5.0, 6.0, 7.0, 8.0}, "pt axis for efficiency histograms"};
+  ConfigurableAxis axisMass{"axisMass", {30, 1.7, 2.0}, "axis of invariant mass of HF candidates"}; // TODO: check if we really need that many bins
+
+  //  Collision filters
+  Filter collisionVtxZFilter = nabs(aod::collision::posZ) < cfgCutVertex;
+  using aodCollisions = soa::Filtered<soa::Join<aod::Collisions, aod::EvSels, aod::Mults>>;
+
+  //  Charged track filters
+  Filter trackFilter = (nabs(aod::track::eta) < cfgCutEta) &&
+                       (aod::track::pt > cfgCutPt) &&
+                       requireGlobalTrackWoPtEtaInFilter();
+  using aodTracks = soa::Filtered<soa::Join<aod::Tracks, aod::TracksDCA, aod::TrackSelection>>;
+
+  //  HF candidate filter
+  //Filter candidateFilter = (aod::hf_selcandidate_d0::isSelD0 >= d_selectionFlagD0 ||
```

Comment:
Hi @saganatt, let me also comment about this point:
In the taskD0, what is done is to use a Partition (at L129), which takes only the D0 candidates satisfying the topological+PID selection, and then at L133 loop only on those candidates, for filling any of the histograms (so in practice this "rejects" the non-selected candidates).
The additional selection at L141 and L144 is used only to differentiate the D0 and D0bar species, since the selection is mass-hypothesis dependent (whether the K is taken as the positive or negative track), and the invariant mass value depends on this hypotheses.
In the current code of taskHFCorrelations.cxx, it seems that at the moment you don't apply any selection on the D0 candidates, right? This would mean filling the correlation containers also for rejected candidates. I would switch to what done in the taskD0 instead.

For what concerns the code location, in view of a better organization of the existing and new codes as proposed by Vit, I think that the code fits better the PWGHF/HFC/Tasks folder.

Reply:
Hi @fcolamar ,
Thank you for your response! I moved the task and gave it a more precise name.

In L133 you don't use non-selected candidates at all, so actually the partition would be equivalent to using a filter? I think this might be Katarina's reasoning.
On the other hand, with such filtering we get different histograms than the taskD0. Only removing the filter and iterating over all candidates as in the current code gives the same results as taskD0. What, in view of your words, is very strange. Let me check if putting partition instead of a filter can impact the results that much.

Reply:
Ciao @saganatt, 
yes, the partition is equivalent to a filter, if I remember well, in the original version of the task there was actually a filter, then when implementing the process switches, the filters were replaced by partitions, to better handle the selection that had to be done on different sets of joined tables, for data and MC processes.

About what you get when running and comparing to the taskD0: this is actually strange, from taskD0 you should get a subset of candidates w.r.t. your task without the filter. Are you sure that you are activating the d_selectionFlagD0 and d_selectionFlagD0bar flags in the json when running the taskD0 (if you use the Run3AnalysisValidation, they are set to 0 in the default json)?

Reply:
Ciao,
I see. Maybe in our task partition will be better as well in the future, I will first try to get concise results with the filter.
I set D0 selection in config_tasks.sh, which in turn sets these flags to 1. But I will re-check it again, thanks for the suggestion.

Reply:
Ciao,
doing that should be ok, then. Strange, let me know. You could also check, from the output of taskD0, the distribution of any cut variable (there are some) and see whether there are entries outside the region alowed by the cut values (e.g. f you didn't modify the cut values, the hCPA histogram should have no entries below 0.80, which is the minimum cut value among the various pT ranges). In case you find entries there, then there's something not working with the candidate selection.

Reply:
I think I found the reason (checks in progress). Before these PRs, when I added more histograms to compare the D0 task with ours one, I accidentally replaced `candidates` instead of the partition. So that explains why we got the results for unfiltered data. Sorry for this, now the filter makes sense :-) , I will just make sure with the validation pipeline the results are indeed the same for the both tasks.